File: src/Moryx.FactoryMonitor.Web/src/app/api/models/Moryx/FactoryMonitor/Endpoints/Models/activity-changed-model.ts
/* tslint:disable */
/* eslint-disable */
import { ActivityClassification as MoryxControlSystemActivitiesActivityClassification } from '../../../../Moryx/ControlSystem/Activities/activity-classification';
import { OrderReferenceModel as MoryxFactoryMonitorEndpointsModelsOrderReferenceModel } from '../../../../Moryx/FactoryMonitor/Endpoints/Models/order-reference-model';
export interface ActivityChangedModel {
  classification?: MoryxControlSystemActivitiesActivityClassification;
  id?: number;
  orderReferenceModel?: MoryxFactoryMonitorEndpointsModelsOrderReferenceModel;
  resourceId?: number;
}

File: src/Moryx.FactoryMonitor.Endpoints/Models/ActivityChangedModel.cs
// Copyright (c) 2025, Phoenix Contact GmbH & Co. KG
// Licensed under the Apache License, Version 2.0

using Moryx.ControlSystem.Activities;
using System.Runtime.Serialization;

namespace Moryx.FactoryMonitor.Endpoints.Models
{

    /// <summary>
    /// Model for the cell information for every ActivityUpdated event.
    /// </summary>
    [DataContract]
    public class ActivityChangedModel
    {
        [DataMember]
        public long Id { get; set; }

        [DataMember]
        public long ResourceId { get; set; }

        [DataMember]
        public ActivityClassification Classification { get; set; }

        //Possible extensions: tracing, ...

        [DataMember]
        public OrderReferenceModel OrderReferenceModel { get; set; }

    }
}

